python_modules/libraries/dagster-aws/dagster_aws/s3/io_manager.py

```diff
@@ -100,11 +104,75 @@ def handle_output(self, context, obj):
         context.add_output_metadata({"uri": MetadataValue.path(path)})
 
 
+class ConfigurablePickledObjectS3IOManager(ConfigurableIOManager):
+    """Persistent IO manager using S3 for storage.
+
+    Serializes objects via pickling. Suitable for objects storage for distributed executors, so long
+    as each execution node has network connectivity and credentials for S3 and the backing bucket.
+
+    Assigns each op output to a unique filepath containing run ID, step key, and output name.
+    Assigns each asset to a single filesystem path, at "<base_dir>/<asset_key>". If the asset key
+    has multiple components, the final component is used as the name of the file, and the preceding
+    components as parent directories under the base_dir.
+
+    Subsequent materializations of an asset will overwrite previous materializations of that asset.
+    With a base directory of "/my/base/path", an asset with key
+    `AssetKey(["one", "two", "three"])` would be stored in a file called "three" in a directory
+    with path "/my/base/path/one/two/".
+
+    Example usage:
+
+    .. code-block:: python
+
+        from dagster import asset, Definitions
+        from dagster_aws.s3 import ConfigurablePickledObjectS3IOManager, S3Resource
+
+
+        @asset
+        def asset1():
+            # create df ...
+            return df
+
+        @asset
+        def asset2(asset1):
+            return asset1[:5]
+
+        defs = Definitions(
+            assets=[asset1, asset2],
+            resources={
+                "io_manager": ConfigurablePickledObjectS3IOManager(
+                    s3_resource=S3Resource(),
+                    s3_bucket="my-cool-bucket",
+                    s3_prefix="my-cool-prefix",
+                )
+            }
+        )
+
+    """
+
+    s3_resource: ResourceDependency[S3Resource]
+    s3_bucket: str = Field(description="S3 bucket to use for the file manager.")
+    s3_prefix: str = Field(
+        default="dagster", description="Prefix to use for the S3 bucket for this file manager."
+    )
+
+    @cached_method
+    def inner_db_io_manager(self) -> PickledObjectS3IOManager:
+        return PickledObjectS3IOManager(
+            s3_bucket=self.s3_bucket,
+            s3_session=self.s3_resource,
```

Comment:
wait what is the behavior here? I think this will not work? we need to call `self.s3_resource.create_client()` no?

Reply:
You're right. The test doesn't catch this since it was passing in an old-style S3 resource. Updated tests + this code accordingly.